src/west/manifest.py

```diff
@@ -144,7 +147,10 @@ def __init__(self, source_file=None, source_data=None,
         projects.
 
         Each element's values are fully initialized; there is no need
-        to consult the defaults field to supply missing values.'''
+        to consult the defaults field to supply missing values.
+
+        Note: The index MANIFEST_PROJECT_INDEX in sequence will hold the
```

Comment:
Guess that is a matter of preference.
When creating a constant, I personally believe it better that people refers to the constant (otherwise why create it), but if writing 0 in the docs, then people might not notice the constant exists and use the value directly.